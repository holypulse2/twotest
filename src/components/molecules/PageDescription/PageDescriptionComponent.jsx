import React from 'react';
import './index.scss'
import Heading from "../../atoms/heading/Heading";
import Text from "../../atoms/text/Text";

const PageDescriptionComponent = ({data}) => {

  return( data &&
    <div>
      <Heading> {data.object.heading} </Heading>
      <Text> {data.object.description} </Text>
    </div>
  )
}

export default PageDescriptionComponent;