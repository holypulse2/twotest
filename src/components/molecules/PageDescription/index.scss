@import '../../../index.scss';

h2 {
    color: $blue;
}

p {
    color: $grey; 
}