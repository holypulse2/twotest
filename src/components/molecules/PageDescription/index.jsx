import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import PageDescriptionComponent from './PageDescriptionComponent';
import { PAGE_DESCRIPTION_COMPONENT } from '../../../constants';

const PageDescription = () => {
  const configuration = useSelector((state) => state.configuration);

  useEffect(() => {
  }, [configuration]);

  if(!configuration.landingPage) return null;

  const data = configuration.landingPage.content.components.find(x => x.id === PAGE_DESCRIPTION_COMPONENT);

  return(data && 
    <PageDescriptionComponent data={data} />
  )

}

export default PageDescription;