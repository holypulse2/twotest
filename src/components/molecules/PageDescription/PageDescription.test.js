import React from 'react';
import { shallow } from 'enzyme';
import PageDescription from './PageDescriptionComponent';

describe('PageDescription', () => {
  const data = { object: { heading: 'test heading', description: 'test description' } }
  
  it('renders correctly', () => {
    const component = shallow(<PageDescription data={data} />);
  
    expect(component).toMatchSnapshot();
  });
});