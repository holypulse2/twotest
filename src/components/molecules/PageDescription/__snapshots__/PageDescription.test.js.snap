// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDescription renders correctly 1`] = `
<div>
  <h2>
     
    test heading
     
  </h2>
  <p>
     
    test description
     
  </p>
</div>
`;
